src/test/java/com/amazon/opendistroforelasticsearch/security/protected_indices/ProtectedIndicesTests.java

```diff
@@ -0,0 +1,578 @@
+import java.util.List;
+
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.amazon.opendistroforelasticsearch.security.test.DynamicSecurityConfig;
+import com.amazon.opendistroforelasticsearch.security.test.SingleClusterTest;
+import com.amazon.opendistroforelasticsearch.security.test.helper.rest.RestHelper;
+import org.apache.http.Header;
+import org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequest;
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.rest.RestStatus;
+import org.junit.Before;
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertTrue;
+
+public class ProtectedIndicesTests extends SingleClusterTest {
+
+    private static final List<String> listOfIndexesToTest = Arrays.asList("logs1", "logs2", "logs3", "no_match");
+    private static final List<String> listOfIndexPatternsToTest = Arrays.asList("*logs*", "logs*", "*lo*");
+    private static final List<String> protectedIndexRoles = Arrays.asList("protected_index_role1", "protected_index_role2");
+    private static final String matchAllQuery = "{\n\"query\": {\"match_all\": {}}}";
+    // This user is mapped to all_access, but is not mapped to any protectedIndexRoles
+    private static final String indexAccessNoRoleUser = "indexAccessNoRoleUser";
+    private static final Header indexAccessNoRoleUserHeader = encodeBasicHeader(indexAccessNoRoleUser, indexAccessNoRoleUser);
+    private static final String generalErrorMessage = String.format("no permissions for [] and User [name=%s, roles=[], requestedTenant=null]", indexAccessNoRoleUser);
+    // This user is mapped to all_access and protected_index_role1
+    private static final String protectedIndexUser = "protectedIndexUser";
+    private static final Header protectedIndexUserHeader = encodeBasicHeader(protectedIndexUser, protectedIndexUser);
+
+    /**
+     * Setup settings loading custom config, users, roles and mappings.
+     * Set the protected indices and protected indices roles.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void setupSettings() throws Exception {
+        // Setup settings
+        Settings adminOnlyIndexSettings = Settings.builder()
+                .put(ConfigConstants.OPENDISTRO_SECURITY_PROTECTED_INDICES_ENABLED_KEY, true)
+                .putList(ConfigConstants.OPENDISTRO_SECURITY_PROTECTED_INDICES_KEY, listOfIndexesToTest)
```

Comment:
This list contains only fixed index names. We need to test also some index-patterns in this setting. Maybe we can replace "no_match" with "no_mat*" .

Reply:
Added a few tests specific to testing patterns (create index & document).